public/app/features/correlations/Forms/QueryEditorField.tsx

```diff
@@ -23,8 +50,53 @@ export const QueryEditorField = ({ dsUid, invalid, error, name }: Props) => {
     }
     return getDataSourceSrv().get(dsUid);
   }, [dsUid]);
+
   const QueryEditor = datasource?.components?.QueryEditor;
 
+  const handleValidation = (value: DataQuery) => {
+    const interval = '1s';
+    const intervalMs = 1000;
+    const id = generateKey();
+    const queries = [{ ...value, refId: 'something' }];
+
+    const transaction: QueryTransaction = {
+      queries,
+      request: {
+        app: CoreApp.Unknown,
+        timezone: 'utc',
+        startTime: Date.now(),
+        interval,
+        intervalMs,
+        targets: queries,
+        range: getDefaultTimeRange(),
+        requestId: 'correlations_' + id,
+        scopedVars: {
+          __interval: { text: interval, value: interval },
+          __interval_ms: { text: intervalMs, value: intervalMs },
+        },
+      },
+      id,
+      done: false,
+    };
+
+    if (datasource) {
+      runRequest(datasource, transaction.request).subscribe((panelData) => {
+        // console.log(panelData)
+        if (!panelData || panelData.state === 'Error' || panelData.series.length === 0) {
+          setIsValidQuery(false);
```

Comment:
Probably it's okay for now but I wonder if it'd make sense to distinguish in the future "This query is not valid" from "This query does not return any results".

Reply:
I agree and suggest to open a follow up task for UX improvements of the validation which contains the following subtasks:
- distinguish "This query is not valid" from "This query does not return any results"
- feedback for users if the query is empty ( see [comment](https://github.com/grafana/grafana/pull/57245#issuecomment-1369007499) above)

Reply:
Opened https://github.com/grafana/grafana/issues/60978